packages/core/src/RenderingEngine/Viewport.ts

```diff
@@ -1159,6 +1163,37 @@ class Viewport implements IViewport {
         viewPlaneNormal,
         focalPoint
       );
+      if (actorEntry?.vtkPlanes && actorEntry?.outline) {
+        const usePolyDataNormalsFilter = false;
+        if (usePolyDataNormalsFilter) {
+          const normals = vtkPolyDataNormals.newInstance();
+          normals.setInputData(actorEntry.polydata);
+          normals.update();
+          const filter = vtkClipClosedSurface.newInstance({
+            clippingPlanes: actorEntry.vtkPlanes,
+            activePlaneId: 2,
+            passPointData: false,
+          });
+          filter.setInputData(normals.getOutputData());
+          filter.setGenerateOutline(true);
+          filter.setGenerateFaces(false);
+          filter.update();
+          const filterData = filter.getOutputData();
+          mapper.setInputData(filterData);
+        } else {
+          const filter = vtkClipClosedSurface.newInstance({
```

Comment:
Also this is getting called a lot, you don't need to `newInstance` it again, i think we can just assign it to the actor and create for the first time and read from it afterwards

Reply:
This also feels a bit weird that we have this logic here, ideally the viewport should trigger an event that say CLIPPING_PLANEES_CHANGED and we listen to that in the `displayTools/surface` and react to that by changing the surface planes